lib/tic_tac_toe/console.ex

```diff
@@ -0,0 +1,17 @@
+defmodule TicTacToe.Console do
+  alias TicTacToe.Display
+  alias TicTacToe.Console
+
+  defstruct state: [], puts: &IO.puts/1, gets: &IO.gets/1
+
+  defimpl Display do
+    def output(%Console{state: state, puts: puts}, message) do
+      puts.(message)
+      state ++ [message]
```

Comment:
Now that you are passing in a struct, you might find it advantageous to return the `Console` struct instead of a list of messages. You could opt to shove thee log of messages into the `Console` struct as well.

Reply:
Will do.